src/SparseEnsembleKalmanInversion.jl

```diff
@@ -115,24 +122,22 @@ function sparse_eki_update(
         # Solve a quadratic programming problem
         u[:, j] = sparse_qp(ekp, v[j, :], cov_vv_inv, H_u, H_g, y[:, j], H_uc = H_uc)
 
-        # Threshold the results if needed
-        if ekp.process.threshold_eki
-            u[ekp.process.uc_idx, j] =
-                u[ekp.process.uc_idx, j] .* (abs.(u[ekp.process.uc_idx, j]) .> ekp.process.threshold_value)
-        end
+        # Prune parameters using threshold
+        u[ekp.process.uc_idx, j] =
+            u[ekp.process.uc_idx, j] .* (abs.(u[ekp.process.uc_idx, j]) .> ekp.process.threshold_value)
 
         # Add small noise to constrained elements of u
-        u[ekp.process.uc_idx, j] += rand(
-            ekp.rng,
-            MvNormal(zeros(size(ekp.process.uc_idx)[1]), ekp.process.reg * I(size(ekp.process.uc_idx)[1])),
-        )
+        if isposdef(ekp.process.reg)
```

Comment:
Is it a convention for our codes to use `isposdef()` for scalars?

Reply:
Not really, I am using it here because it guarantees the same precision needed to draw elements from a distribution (there is an internal isposdef check to the covariance matrix)